sys/pm_callouts.c

```diff
@@ -546,26 +509,12 @@ void send_icmp_blocked_packet(portmaster_packet_info* packetInfo, void* original
         // Calculate checksum for the icmp packet
         calc_ipv4_checksum(icmpPacket, packetLength, TRUE);
 
-        // Create the net buffer list for the new ICMP packet
-        status = wrap_packet_data_in_nb(icmpPacket, packetLength, &injectNBL);
-        if (!NT_SUCCESS(status)) {
-            ERR("ICMP Port unreachable -> wrap_packet_data_in_nb failed: 0x%x", status);
-            portmaster_free(icmpPacket);
-            return;
-        }
-
-        if(packetInfo->direction == 1 || is_ipv4_loopback(packetInfo->remoteIP[0]) || useLocalHost) {
-            // InjectNetworkReceive does not work for loopback packet but send works!?
-            status = FwpsInjectNetworkSendAsync(inject_out4_handle, 0, 0, UNSPECIFIED_COMPARTMENT_ID,
-                    injectNBL, free_after_inject,
-                    icmpPacket);
+        // Reverse diraction and inject packet
+        reverseDirection = packetInfo->direction == 1 ? 0 : 1;
```

Comment:
Should the direction regard the `useLocalHost` flag?

Reply:
it will automatically  switch to send in the inject_packet function if the ip is loalhost